src/test/java/edu/project1/ConsoleHangmanTest.java

```diff
@@ -0,0 +1,224 @@
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.Serializable;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ConsoleHangmanTest {
+
+    private ConsoleHangman game;
+
+    @Plugin(name = "TestAppender", category = "Core", elementType = "appender", printObject = true)
+    public static class TestAppender implements Appender {
+        private static final List<String> log = new LinkedList<>();
+        private static final PatternLayout layout = PatternLayout.newBuilder().withPattern("%m%n").build();
+
+        @PluginFactory public static TestAppender createAppender() {
+            return new TestAppender();
+        }
+
+        @Override public void append(LogEvent event) {
+            log.add(layout.toSerializable(event));
+        }
+
+        public static String getLog() {
+            return String.join("", log);
+        }
+
+        public static void clearLog() {
+            log.clear();
+        }
+
+        @Override public String getName() {
+            return "TestAppender";
+        }
+
+        @Override public Layout<? extends Serializable> getLayout() {
+            return layout;
+        }
+
+        @Override public boolean ignoreExceptions() {
+            return false;
+        }
+
+        @Override public ErrorHandler getHandler() {
+            return null;
+        }
+
+        @Override public void setHandler(ErrorHandler handler) {
+
+        }
+
+        @Override public State getState() {
+            return State.STARTED;
+        }
+
+        @Override public void initialize() {
+
+        }
+
+        @Override public void start() {
+
+        }
+
+        @Override public void stop() {
+
+        }
+
+        @Override public boolean isStarted() {
+            return true;
+        }
+
+        @Override public boolean isStopped() {
+            return false;
+        }
+    }
+
+    @BeforeEach void setUp() {
+        game = new ConsoleHangman();
+        TestAppender.clearLog();
+    }
+
+    @AfterEach void restoreStreams() {
+        System.setIn(System.in);
+    }
+
+    @Test
+    void gameShouldNotStartWithIncorrectWordLength() {  // Игра не должна начинаться, если длина загаданного слова некорректна (в данном случае пустая строка)
+        // Arrange
+        game.setSession(new Session("", 5));
+
+        // Act
+        game.run();
+
+        // Assert
+        assertTrue(TestAppender.getLog().contains("The chosen word has an incorrect length. Exiting the game."));
+    }
+
+    @Test
+    void gameShouldEndWithDefeatAfterMaxAttempts() { // Игра должна завершиться проигрышем после максимального количества попыток (в данном случае 1 попытка)

```

Comment:
`BUG` в ржаве методы тестов принято называть в snake_case или полу snake_case.
Например:
```suggestion
    void gameShouldEndWithDefeat_AfterMaxAttempts() { // Игра должна завершиться проигрышем после максимального количества попыток (в данном случае 1 попытка)

```